### DISCUSSION TITLE
asus xg-c100f 10g sfp+ network adapter

### DISCUSSION THREAD

**Discussion by PatcoreShare** [ipxe/ipxe#1039] [Q&A]
I have an issue with starting up a computer equipped with the ASUS XG-C100F 10G SFP+ Network Adapter.
![image](https://github.com/ipxe/ipxe/assets/11398415/e3ea1c8a-10b8-46d9-8a2e-a07d2ebf33c1)
The settings appear to be correct.
![image](https://github.com/ipxe/ipxe/assets/11398415/8abf10ef-d410-4a9f-a36a-68546571613d)
However, I am unable to use the 'ping' command to ping another machine.
![image](https://github.com/ipxe/ipxe/assets/11398415/be1d9682-4679-4aa2-a960-991fbb1e694e)
Ifstat 
![image](https://github.com/ipxe/ipxe/assets/11398415/e1ae5313-5f90-40ae-ad5d-940f8835e519)
"Network card: ASUS XG-C100F 10G SFP+ Network Adapter. 

Another computer equipped with a different network card is functioning correctly. How can I resolve my issue? Where should I look for errors?"

**Comment by NiKiZe** [comment]
iPXE here uses the NII driver, which is provided by the efi firmware. You have both RXE and TXE, which is totally normal in a busy network. But I would still look into if packets are sent correctly (tcpdump/wireshark), maybe there is a MTU issue or missconfiguration.

**Reply by PatcoreShare** [reply]
Do I need to search for something specific?
```
asd@ipxe-dhcp:~$ sudo tcpdump -i ens160 host 10.0.101.12
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on ens160, link-type EN10MB (Ethernet), snapshot length 262144 bytes
07:38:00.509896 IP 10.0.101.12.1966 > ipxe-dhcp.tftp: TFTP, length 51, RRQ "ipxe.efi" octet tsize 0 blksize 1468 windowsize 4
07:38:00.510988 IP ipxe-dhcp.39088 > 10.0.101.12.1966: UDP, length 29
07:38:00.511083 IP 10.0.101.12.1966 > ipxe-dhcp.39088: UDP, length 30
07:38:00.668958 IP 10.0.101.12.1967 > ipxe-dhcp.tftp: TFTP, length 43, RRQ "ipxe.efi" octet blksize 1468 windowsize 4
07:38:00.670003 IP ipxe-dhcp.36818 > 10.0.101.12.1967: UDP, length 15
07:38:00.670112 IP 10.0.101.12.1967 > ipxe-dhcp.36818: UDP, length 4
07:38:00.670144 IP ipxe-dhcp.36818 > 10.0.101.12.1967: UDP, length 1472
……..
07:38:00.746834 IP ipxe-dhcp.36818 > 10.0.101.12.1967: UDP, length 1472
07:38:00.746929 IP 10.0.101.12.1967 > ipxe-dhcp.36818: UDP, length 4
07:38:00.746944 IP ipxe-dhcp.36818 > 10.0.101.12.1967: UDP, length 60
07:38:00.747024 IP 10.0.101.12.1967 > ipxe-dhcp.36818: UDP, length 4
07:38:00.794110 IP 10.0.101.12.1968 > ipxe-dhcp.tftp: TFTP, length 43, RRQ "autoexec.ipxe" octet tsize 0 windowsize 4
07:38:00.795034 IP ipxe-dhcp.42809 > 10.0.101.12.1968: UDP, length 19
^C
1382 packets captured
1382 packets received by filter
0 packets dropped by kernel
```

In Legacy mode, it works with undionly.kpxe.  
So it's not a network issue; where can I look for a solution? Did I create ipxe.pxe incorrectly

**Reply by NiKiZe** [reply]
The small/UDP packets already looks fine from the screenshots, the issue is with the http request/response/connection

**Reply by PatcoreShare** [reply]
TFTP is the same ![image](https://github.com/ipxe/ipxe/assets/11398415/a812988b-3105-4232-a745-84e8de123b22) 


but in Legacy mode, it works with undionly.kpxe.

**Reply by NiKiZe** [reply]
Did you check what your MTU settings are?
Your server seems to be on a different subnet from the client, so also check routing.

Maybe the efi driver (provided by the manufacturer) have a MTU bug which drops packets that is at the limit of normal MTU,  or maybe even has incorrect internal reports of its MTU.

What you should check for in tcpdump is packet sizes close to the 1500 bytes and see if/where they might be dropped.

Have you contacted the manufacturer?
